functions/send_encounters/main.py

```diff
@@ -46,18 +46,19 @@ def send_encounters(request):
     app_version = request_data[KEY_APP_VERSION]
     lang = request_data[KEY_LANG]
     proof = request_data[KEY_PROOF]
-    encounters = request_data[KEY_ENCOUNTERS]
+    encounters = request_data.get(KEY_ENCOUNTERS)
 
     user_entity = _get_user_entity(user_id)
 
     if not user_entity:
         return jsonify({"status": "failed", "message": "unauthorized"}), 401
 
-    upload_id = secrets.token_hex(32)
+    if encounters:
+        upload_id = secrets.token_hex(32)
 
-    _save_encounter_uploads_to_datastore(user_id, upload_id, proof)
-    if not _save_encounters_to_bigquery(user_id, upload_id, encounters):
-        return jsonify({"status": "failed", "message": "Internal error"}), 500
+        _save_encounter_uploads_to_datastore(user_id, upload_id, proof)
+        if not _save_encounters_to_bigquery(user_id, upload_id, encounters):
```

Comment:
hmm, z tego co rozmawialismy to `upload_id` powinien sie zapisac nawet jesli nie ma zadnych `encounters`

Reply:
OK, tylko... po co? To będzie potencjalnie mylące że coś będzie w Datastore a nie w BigQuery. 
Może dodać flagę `empty` do datastore'a? Albo count?

Reply:
Po to że jak IS wysyla prosbe do user'a zeby przeslal dane, user przesyla zero danych i nie zostaje po tym slad to wtedy IS bedzie go dalej pingał... jak zostanie slad ze przeslal zero encounters to mu odpuszcza

Reply:
OK, słusznie - dodam.